bin/citrea/tests/e2e/mod.rs

```diff
@@ -3387,3 +3388,111 @@ async fn test_sequencer_commitment_threshold() {
 
     seq_task.abort();
 }
+
+#[tokio::test(flavor = "multi_thread")]
+async fn test_sequencer_fill_missing_da_blocks() -> Result<(), anyhow::Error> {
+    // citrea::initialize_logging(tracing::Level::INFO);
+
+    let storage_dir = tempdir_with_children(&["DA", "sequencer"]);
+    let da_db_dir = storage_dir.path().join("DA").to_path_buf();
+    let sequencer_db_dir = storage_dir.path().join("sequencer").to_path_buf();
+
+    let (seq_port_tx, seq_port_rx) = tokio::sync::oneshot::channel();
+
+    let da_db_dir_cloned = da_db_dir.clone();
+    let seq_task = tokio::spawn(async {
+        start_rollup(
+            seq_port_tx,
+            GenesisPaths::from_dir(TEST_DATA_GENESIS_PATH),
+            None,
+            NodeMode::SequencerNode,
+            sequencer_db_dir,
+            da_db_dir_cloned,
+            DEFAULT_MIN_SOFT_CONFIRMATIONS_PER_COMMITMENT,
+            true,
+            None,
+            None,
+            Some(true),
+            DEFAULT_DEPOSIT_MEMPOOL_FETCH_LIMIT,
+        )
+        .await;
+    });
+
+    let seq_port = seq_port_rx.await.unwrap();
+    let seq_test_client = init_test_rollup(seq_port).await;
+
+    seq_test_client.send_publish_batch_request().await;
+    wait_for_l2_block(&seq_test_client, 1, None).await;
+
+    let da_service = MockDaService::new(MockAddress::from([0; 32]), &da_db_dir);
+
+    let to_be_filled_da_block_count = 5;
+    let latest_da_block = 1 + to_be_filled_da_block_count;
+    // publish da blocks back to back
+    for _ in 0..to_be_filled_da_block_count {
+        da_service.publish_test_block().await.unwrap();
+    }
+    wait_for_l1_block(&da_service, latest_da_block, None).await;
+
+    let mut first_filler_l2_block = 1;
+    // wait for sequencer to notice the gap with the da
+    let mut retry_limit = 5;
+    loop {
```

Comment:
This loop should not be needed.

Once an L2 is published, the force handler and the regular handler should see that there are L1 blocks that were not filled. This will happen if you wait long enough which is 2 seconds by default. You can add a sleep on line 3436.

Reply:
I am aware of that, but it feels flaky to me, as there is a possibility of changing the default L1 poll interval, which could cause this test to fail. Do you think it should be replaced with sleep despite that?

Reply:
Yes.... the sleep is configurable in `SequencerConfig` in the test so you can make the sleep you use more or equal to the duration of the sequencer L1 poll interval.

Reply:
That is quite correct :). Here is the update: https://github.com/chainwayxyz/citrea/pull/861/commits/1b576570b7ae15c140cc78bd0d5fcc2d062888cc